File: javaClient/src/main/java/cb/bots/AiFactory.java
package cb.bots;

public class AiFactory {
     public static BotAi create(String type) {
         switch (type) {
             case "fill":
                 return new FillBot();
             case "user":
                 return new UserBot();
             default:
                 return new RandomBot();
         }
     }
}

File: pythonClient/Bots/aiFactory.py
from Bots.bot_ai import BotAI
from Bots.fill_ai import FillAI
from Bots.random_ai import RandomAI
from Bots.user_ai import UserAI

ai_bots = {
    "random": RandomAI,
    "fill": FillAI,
    "user": UserAI
}


def AiFactory(bot_selection="random") -> BotAI:
    return ai_bots[bot_selection]()
